### PR TITLE
Deploy user supplied confs without adding defaults

### PR DESCRIPTION
**Pull request by rschlussel**
Changes behavior of `configuration deploy` to deploy any user defined
config files as is, without adding in default values for undefined
"required properties".  If no configuration file is present for
'node.properties', 'config.properties', or 'jvm.config', presto-admin
will deploy a default configuration file.

This change makes it possible for a user to remove properties that
are no longer valid in presto (for example if the user has upgraded to a
different presto version where a property that had been required now no
longer exists).

It also means that to change some config properties to non-default
values, the user must write out the whole configuration file

Testing: make test-all

@ebd2 @anusudarsan  could you please review?
@brian-rickman fyi, and if you have any comments about usability

### CODE REVIEW

File: prestoadmin/config.py [hunk 1]
```diff
@@ -71,6 +71,26 @@ def json_to_string(conf):
     return json.dumps(conf, indent=4, separators=(',', ':'))
 
 
+def write_conf_to_file(value, path):
```
Comment (ebd2) [inline prestoadmin/config.py]: I propose rename 'value' to conf, as it isn't a single value.

File: prestoadmin/node.py [hunk 2]
```diff
@@ -40,21 +41,29 @@ def get_conf(self):
...
 
-        defaults = self.build_defaults()
-        config.fill_defaults(conf, defaults)
         self.validate(conf)
         return conf
 
     @abstractmethod
     def _get_conf_dir(self):
-        pass
+        raise NotImplementedError()
```
Comment (ebd2) [inline prestoadmin/node.py]: To further my python knowledge, why the change here? You shouldn't be able to instantiate a Node subclass that doesn't implement this anyway, right?

File: prestoadmin/config.py (hunk 1 above)
Comment (ebd2) [inline prestoadmin/config.py]: Can we please get a comment that notes that this only works for flat configs so that when somebody tries to use it for something hierarchical, they know that they need to add functionality?

**Comment by ebd2** [comment]
This looks generally good to me. I'll wait to push until after you do and I'll sort out any merge issues that arise. My money is on there being several.

File: prestoadmin/node.py (hunk 2 above)
Comment (rschlussel) [inline prestoadmin/node.py]: not needed.  I'm switching it back.  I think I switched it because pycharm gives a type check warning, but it's not a correct warning..

**Comment by anusudarsan** [comment]
lgtm
